test/e2e/nodepool_kv_advanced_multinet_test.go

```diff
@@ -0,0 +1,328 @@
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if len(passthroughService.Spec.Ports) == 0 {
+		return fmt.Errorf("missing passthrough service port")
+	}
+	nodePort := passthroughService.Spec.Ports[0].TargetPort.String()
+
+	output, err := k.configureDNAT(t, machineAddress, nodePort)
+	if err != nil {
+		return fmt.Errorf("failed configuring nat at dnmasq proxy pod: %w: %s", err, string(output))
+	}
+	return nil
+}
+
+// TeardownInfra delete dnsmasq and nad that are deployed at default namespace
+// to be privileged
+func (k KubeVirtAdvancedMultinetTest) TeardownInfra(t *testing.T) error {
+	infraClient, err := k.infra.DiscoverClient()
+	if err != nil {
+		return nil
+	}
+
+	errs := []error{}
+
+	nad, err := k.infra.ComposeOVNKLayer2NAD("default")
+	if err != nil {
+		errs = append(errs, err)
+	} else if err := infraClient.Delete(k.infra.Ctx(), nad); err != nil {
+		errs = append(errs, err)
+	}
+
+	if err := infraClient.Delete(k.infra.Ctx(), k.composeDNSMasqPod()); err != nil {
+		errs = append(errs, err)
+	}
+
+	return errors.Join(errs...)
+}
+
+func (k KubeVirtAdvancedMultinetTest) configureDNAT(t *testing.T, machineAddress, nodePort string) (string, error) {
+	dnsmasqPod := k.composeDNSMasqPod()
+	command := fmt.Sprintf(`
+apk update
+apk add iptables
+iptables -t nat -A PREROUTING -p tcp --dport %[2]s -j DNAT --to-destination %[1]s:%[2]s
+`, machineAddress, nodePort)
+	infraClient, err := k.infra.DiscoverClient()
+	if err != nil {
+		return "", err
+	}
+	return e2eutil.RunCommandInPod(k.infra.Ctx(), infraClient, dnsmasqPod.Name, dnsmasqPod.Namespace, []string{"/bin/sh", "-c", command}, dnsmasqPod.Spec.Containers[0].Name)
+}
+
+func (k KubeVirtAdvancedMultinetTest) firstMachineAddress() (string, error) {
+	machineList := capkv1alpha1.KubevirtMachineList{}
+	namespace := manifests.HostedControlPlaneNamespace(k.infra.HostedCluster().Namespace, k.infra.HostedCluster().Name)
+	if err := k.infra.MGMTClient().List(k.infra.Ctx(), &machineList, client.InNamespace(namespace),
+		client.MatchingLabels{
+			clusterapiv1beta1.MachineDeploymentNameLabel: k.nodePoolName,
+		}); err != nil {
+		return "", err
+	}
+	if len(machineList.Items) == 0 {
+		return "", fmt.Errorf("first kubevirt machine not found")
+	}
+
+	internalAddress := ""
+	for _, address := range machineList.Items[0].Status.Addresses {
+		if address.Type == "InternalIP" { //TODO use constant
+			internalAddress = address.Address
+		}
+	}
+	if internalAddress == "" {
+		return "", fmt.Errorf("missing internal addres at kubevirt machine")
+	}
+	return internalAddress, nil
+}
+
+func (k KubeVirtAdvancedMultinetTest) composeDNSMasqPod() *corev1.Pod {
+	podName := k.infra.NADName() + "-dnsmasq"
+	return &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+			Name:      podName,
+			Annotations: map[string]string{
+				"k8s.v1.cni.cncf.io/networks": fmt.Sprintf(`[{"interface":"%s","name":"%s","namespace":"%s"}]`, k.ifaceName, k.infra.NADName(), "default"),
```

Comment:
Any reason not to use `json.Marshal`?

Reply:
An anonymous struct is easy enough to put in, up to you - certainly easier to handle marshal errors than whatever happens when string-format doesn't create valid JSON